core/src/main/java/org/keycloak/sdjwt/consumer/JwtVcMetadataTrustedSdJwtIssuer.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.sdjwt.consumer;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import org.keycloak.common.VerificationException;
+import org.keycloak.crypto.SignatureVerifierContext;
+import org.keycloak.sdjwt.IssuerSignedJWT;
+import org.keycloak.sdjwt.JwkParsingUtils;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A trusted Issuer for running SD-JWT VP verification.
+ *
+ * <p>
+ * This implementation targets issuers exposing verifying keys on a normalized JWT VC Issuer metadata endpoint.
+ * </p>
+ *
+ * @author <a href="mailto:Ingrid.Kamga@adorsys.com">Ingrid Kamga</a>
+ * @see <a href="https://datatracker.ietf.org/doc/html/draft-ietf-oauth-sd-jwt-vc-05#name-issuer-signed-jwt-verificat">
+ * JWT VC Issuer Metadata
+ * </a>
+ */
+public class JwtVcMetadataTrustedSdJwtIssuer implements TrustedSdJwtIssuer {
+
+    private static final String JWT_VC_ISSUER_END_POINT = "/.well-known/jwt-vc-issuer";
+
+    private final Pattern issuerUriPattern;
+    private final HttpDataFetcher httpDataFetcher;
+
+    /**
+     * @param issuerUri a trusted issuer URI
+     */
+    public JwtVcMetadataTrustedSdJwtIssuer(String issuerUri, HttpDataFetcher httpDataFetcher) {
+        try {
+            validateHttpsIssuerUri(issuerUri);
+        } catch (VerificationException e) {
+            throw new IllegalArgumentException(e);
+        }
+
+        // Build a Regex pattern to only match the argument URI
+        this.issuerUriPattern = Pattern.compile(Pattern.quote(issuerUri));
```

Comment:
What does the pattern check, beside that its a valid uri? Wont URI.create(..) be an easier solution? It could also reduce the subsequent string handling(like startsWith("https").

Reply:
The pattern here is not intended for validating the URI as an URI. That is left to a concrete HTTP client. Instead, the class allows specifying a regex pattern to cover many trusted Issuer URIs at once; for example, `.*.google.com`. I would appreciate any further guidance on this.